adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -808,6 +807,10 @@ func createRoute(params *routeCreateParams) *routev3.Route {
 		Value:   luaMarshelled.Bytes(),
 	}
 
+	pathRegex := xWso2Basepath
+	if params.rewritePath != "" {
+		pathRegex = routePath
```

Comment:
check path params.

Reply:
routepath is a regex. so it's correct. however, the rewrite path should not include path params. hence added validation for it.